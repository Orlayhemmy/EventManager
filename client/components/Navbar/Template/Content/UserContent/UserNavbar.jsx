import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description NavBar component
 */
export default class UserNavbar extends React.Component {
  render() {
    const {
      navbarProps: { path, logout }
    } = this.props;
    const navbarSecondary = (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <Link
            id="dashboard"
            to="/dashboard"
            className={path === '/dashboard' ? 'nav-link active' : 'nav-link'}
          >
            <span className="nav-link-text">Dashboard</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            id="add-event"
            to="/add-event"
            className={path === '/add-event' ? 'nav-link active' : 'nav-link'}
          >
            <span className="nav-link-text">add event</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            id="profile"
            to="/profile"
            className={path === '/profile' ? 'nav-link active' : 'nav-link'}
          >
            <span className="nav-link-text">profile</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/" id="logout" onClick={logout} className="nav-link">
            <span className="nav-link-text">logout</span>
          </Link>
        </li>
      </ul>
    );

    const navbarMain = (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item">
          <Link
            to="/"
            className={path === '/' ? 'nav-link active' : 'nav-link'}
          >
            <span id="home" className="nav-link-text">
              Home
            </span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            id="view-centers"
            to="/view-centers"
            className={
              path === '/view-centers' ? 'nav-link active' : 'nav-link'
            }
          >
            <span className="nav-link-text">centers</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            id="about"
            to="/about"
            className={path === '/about' ? 'nav-link active' : 'nav-link'}
          >
            <span className="nav-link-text">about us</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link
            id="dashboard"
            to="/dashboard"
            className={path === '/dashboard' ? 'nav-link active' : 'nav-link'}
          >
            <span className="nav-link-text">Dashboard</span>
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/" id="logout" onClick={logout} className="nav-link">
            <span className="nav-link-text">logout</span>
          </Link>
        </li>
      </ul>
    );
    return path === '/' || path === '/about' ? navbarMain : navbarSecondary;
  }
}
