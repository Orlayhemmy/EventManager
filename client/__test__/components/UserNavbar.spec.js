import React from 'react';
import UserNavbar from '../../components/Navbar/Template/Content/UserContent/UserNavbar';
import { navbarProps } from './MockData';

describe('User navbar', () => {
  const wrapper = shallow(<UserNavbar navbarProps={navbarProps} />);
  it('should render the component without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
