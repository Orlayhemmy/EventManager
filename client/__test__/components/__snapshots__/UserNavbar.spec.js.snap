// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User navbar should render the component without errors 1`] = `
<ul
  className="navbar-nav ml-auto"
>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link active"
      replace={false}
      to="/"
    >
      <span
        className="nav-link-text"
        id="home"
      >
        Home
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      id="view-centers"
      replace={false}
      to="/view-centers"
    >
      <span
        className="nav-link-text"
      >
        centers
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      id="about"
      replace={false}
      to="/about"
    >
      <span
        className="nav-link-text"
      >
        about us
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      id="dashboard"
      replace={false}
      to="/dashboard"
    >
      <span
        className="nav-link-text"
      >
        Dashboard
      </span>
    </Link>
  </li>
  <li
    className="nav-item"
  >
    <Link
      className="nav-link"
      id="logout"
      onClick={[MockFunction]}
      replace={false}
      to="/"
    >
      <span
        className="nav-link-text"
      >
        logout
      </span>
    </Link>
  </li>
</ul>
`;
